src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/SqlColumnEncryptionAzureKeyVaultProvider.cs

```diff
@@ -282,173 +276,69 @@ public override byte[] DecryptColumnEncryptionKey(string masterKeyPath, string e
         public override byte[] EncryptColumnEncryptionKey(string masterKeyPath, string encryptionAlgorithm, byte[] columnEncryptionKey)
         {
             // Validate the input parameters
-            this.ValidateNonEmptyAKVPath(masterKeyPath, isSystemOp: true);
-
-            if (null == columnEncryptionKey)
-            {
-                throw new ArgumentNullException(Constants.AeParamColumnEncryptionKey, Strings.NullCek);
-            }
-
-            if (0 == columnEncryptionKey.Length)
-            {
-                throw new ArgumentException(Strings.EmptyCek, Constants.AeParamColumnEncryptionKey);
-            }
+            ValidateNonEmptyAKVPath(masterKeyPath, isSystemOp: true);
+            ValidateNotNullOrWhitespace(encryptionAlgorithm, nameof(encryptionAlgorithm));
+            ValidateEncryptionAlgorithm(encryptionAlgorithm, isSystemOp: true);
+            ValidateNotNull(columnEncryptionKey, nameof(columnEncryptionKey));
+            ValidateNotEmpty(columnEncryptionKey, nameof(columnEncryptionKey));
 
-            // Validate encryptionAlgorithm
-            this.ValidateEncryptionAlgorithm(ref encryptionAlgorithm, isSystemOp: false);
-
-            // Validate whether the key is RSA one or not and then get the key size
-            int keySizeInBytes = GetAKVKeySize(masterKeyPath);
+            // Also validates whether the key is RSA one or not and then get the key size
+            KeyCryptographer.AddKey(masterKeyPath);
+            int keySizeInBytes = KeyCryptographer.GetKeySize(masterKeyPath);
 
             // Construct the encryptedColumnEncryptionKey
             // Format is 
             //          version + keyPathLength + ciphertextLength + ciphertext + keyPath + signature
             //
             // We currently only support one version
-            byte[] version = new byte[] { firstVersion[0] };
+            byte[] version = new byte[] { s_firstVersion[0] };
```

Comment:
`s_firstVersion` is alredy a byte array. No need to unwrap it and then put it back into a new byte array.

From the member variables above:
`private readonly static byte[] s_firstVersion = new byte[] { 0x01 };`